autoload/codefmt.vim

```diff
@@ -228,6 +229,54 @@ if !exists('s:autopep8')
   call codefmtlib#AddDefaultFormatter(s:autopep8)
 endif
 
+
+" Formatter: js-beautify
+if !exists('s:js_beautify')
+  let s:js_beautify = {
+      \ 'name': 'js_beautify',
+      \ 'setup_instructions': 'Install js-beautify ' .
+          \ '(https://www.npmjs.com/package/js-beautify).'}
+
+  function s:js_beautify.IsAvailable() abort
+    return executable(s:plugin.Flag('js_beautify_executable'))
+  endfunction
+
+  function s:js_beautify.AppliesToBuffer() abort
+    return &filetype is# 'css' || &filetype is# 'html' || &filetype is# 'json' ||
+        \ (&filetype is# 'javascript' &&
+            \ s:plugin.Flag('javascript_formatter') is# 'js_beautify')
+  endfunction
+
+  ""
+  " Reformat the current buffer with js-beautify or the binary named in
+  " @flag(js_beautify_executable), only targeting the range between {startline} and
+  " {endline}.
+  " @throws ShellError
+  function s:js_beautify.FormatRange(startline, endline) abort
+    let l:cmd = [ s:plugin.Flag('js_beautify_executable'),
+                \'--type', &filetype,
```

Comment:
i did not thought about the case of calling it with specifying formatter manually, i will fix it